acapy/plugins/firebase_push_notifications/firebase_push_notifications/v1_0/manager.py

```diff
@@ -0,0 +1,112 @@
+from datetime import timedelta
+from dateutil import parser
+import logging
+import json
+import os
+import requests
+
+import google.auth.transport.requests
+from google.oauth2 import service_account
+
+from aries_cloudagent.messaging.util import time_now, datetime_now
+from .models import FirebaseConnectionRecord
+
+from .constants import SCOPES, BASE_URL, ENDPOINT_PREFIX, ENDPOINT_SUFFIX
+
+PROJECT_ID = os.environ.get('FCM_PROJECT_ID')
+FCM_ENDPOINT = ENDPOINT_PREFIX + PROJECT_ID + ENDPOINT_SUFFIX
+FCM_URL = BASE_URL + '/' + FCM_ENDPOINT
+MAX_SEND_RATE_MINUTES = int(os.environ.get('FCM_MAX_SEND_RATE_MINUTES'))
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_access_token():
+    """Retrieve a valid access token that can be used to authorize requests."""
+    credentials = service_account.Credentials.from_service_account_info(
+        json.loads(os.environ.get('FCM_SERVICE_ACCOUNT')), scopes=SCOPES)
+    request = google.auth.transport.requests.Request()
+    credentials.refresh(request)
+    return credentials.token
+
+
+async def send_message(profile, connection_id):
+    headers = {
```

Comment:
Should functions in python have a description of what they do as the first line?
```python
""" This fn is super cool because of what it does"""